rest.go

```diff
@@ -0,0 +1,333 @@
+		buffer.WriteString(fmt.Sprintf(s, key, getContextKey(key)))
+	}
+	for _, qp := range queryParams {
+		s = "	%s := r.URL.Query().Get(\"%s\")\n"
+		buffer.WriteString(fmt.Sprintf(s, getQueryVar(qp), qp))
+	}
+	return buffer.String()
+}
+
+const errBoiler = `if err != nil {
+		http.Error(w, err.Error(), getStatus(err))
+		return
+	}`
+
+func genGet(handler string, r *route) string {
+	var buffer bytes.Buffer
+	buffer.WriteString(getHandlerHead(handler, r.keys, r.queryParams))
+	params := strings.Join(append(r.keys, r.queryParams...), ", ")
+	s := `	result, err := rh.storer.%s(%s)
+	%s
+	render.JSON(w, r, result)
+}` + "\n\n"
+	buffer.WriteString(fmt.Sprintf(s, handler, params, errBoiler))
+	return buffer.String()
+}
+
+func genDelete(handler string, r *route) string {
+	var buffer bytes.Buffer
+	buffer.WriteString(getHandlerHead(handler, r.keys, r.queryParams))
+	params := strings.Join(append(r.keys, r.queryParams...), ", ")
+	s := `	if err := rh.storer.%s(%s); err != nil {
+		http.Error(w, err.Error(), getStatus(err))
+		return
+	}
+	render.NoContent(w, r)
+}` + "\n\n"
+	buffer.WriteString(fmt.Sprintf(s, handler, params))
+	return buffer.String()
+}
+
+const payloadBoiler = `	if err := decodeJSON(r.Body, &payload); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}` + "\n"
+
+func genPutPost(handler string, r *route, post bool) string {
+	var buffer bytes.Buffer
+	buffer.WriteString(getHandlerHead(handler, r.keys, r.queryParams))
+	p := append(r.keys, r.queryParams...)
+	p = append(p, "payload")
+	params := strings.Join(p, ", ")
+	payloadType := r.putPayloadType
+	status := ""
+	if post {
+		payloadType = r.postPayloadType
+		status = "\nrender.Status(r, http.StatusCreated)"
+	}
+
+	s := `	var payload %s
+	%s
+	result, err := rh.storer.%s(%s)
+	%s %s
+	render.JSON(w, r, result)
+}` + "\n\n"
+	str := fmt.Sprintf(s,
+		payloadType,
+		payloadBoiler,
+		handler,
+		params,
+		errBoiler,
+		status)
+	buffer.WriteString(str)
+	return buffer.String()
+}
+
+func genPut(handler string, r *route) string {
+	return genPutPost(handler, r, false)
+}
+
+func genPost(handler string, r *route) string {
+	return genPutPost(handler, r, true)
+}
+
+func genNewRestHandler(r routes) string {
+	var buffer bytes.Buffer
+	buffer.WriteString(`// NewRestHandler creates a new Handler persisting data to Storer.
+func NewRestHandler(s Storer, m Middleware) RestHandler {
+	r := chi.NewRouter()
+	r.Use(m.Root()...)
+	rh := RestHandler{s, r}` + "\n\n")
+	buffer.WriteString(genRoutes(r))
+	buffer.WriteString(`return rh
+}` + "\n\n")
+	return buffer.String()
+}
+
+func genRoutes(r routes) string {
+	var buffer bytes.Buffer
+	for _, path := range r.paths {
+		str := fmt.Sprintf("r.Route(\"%s\", func(r chi.Router) {\n", path)
+		buffer.WriteString(str)
+		for _, key := range r.content[path].keys {
+			ctxKey := getContextKey(key)
+			str = fmt.Sprintf("r.Use(makeContextSaver(%s, \"%s\"))\n", ctxKey, key)
```

Comment:
\`...\` In fact, for consistency, it might be a good idea to use backquotes for most code snippets, whether they need it or not.

Reply:
The problem with this approach is that it escapes all `\n` - I've only replaced it were it simplifies the code.